tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/PublishCoreSetupBinaries.cs

```diff
@@ -28,7 +28,9 @@ public class PublishCoreSetupBinaries : Task
         //   dotnet-runtime-rhel.7-x64.2.0.0-preview2-25401-9.tar.gz
         //   dotnet-runtime-2.0.0-preview2-25401-9-rhel.7-x64.tar.gz
         // the "semver" capture would be 2.0.0-preview2-25401-9 in this case.
-        protected virtual string VersionMatchRegex => @"(\.|-)(?'semver'[0-9]+\.[0-9]+\.[0-9]+(-[A-Za-z0-9]+(-|\.)[0-9]+(-|\.)[0-9]+((-|\.)[0-9]+)?)?)";
+        protected virtual string VersionMatchRegex => @"(\.|-)(?'semver'[0-9]+\.[0-9]+\.[0-9]+(([-.])[A-Za-z0-9]+)*)";
+
+        protected string[] BadAtoms = new[] { "-x64", ".x64", ".tar", ".gz", "-rhel.7", "-rhel.8", ".rhel.7", ".rhel.8", "-centos.7", "-centos.8", ".centos.7", ".centos.8", "-linux", ".linux", ".ubuntu.18.04", "-ubuntu.18.04", "-debian.9", ".debian.9" };
```

Comment:
Chop please! (If it stays here.)

Would be much better to take these via an input. Since they're compiled every single time we build from source, it's less of an issue in that respect, but if we take it as an input there's a greater chance we could share this list with some other infra that lists this kind of thing: particularly RIDs. It seems like hooking up the current RID into this would replace the hard-coded list of various RIDs we happen to build with. (At least that looks like the commonality....)

Is there a way to know from a specific type of error that you have to modify this list? Is there a useful comment that we could add here and/or elsewhere?